src/util/request.js

```diff
@@ -152,6 +161,34 @@ export const apiPaths = {
   audits: (query) => `/v1/audits${queryString(query)}`
 };
 
+
+
+////////////////////////////////////////////////////////////////////////////////
+// SENDING REQUESTS
+
+const httpMethods = {};
+for (const prop of ['get', 'delete']) {
+  const method = prop.toUpperCase();
+  // eslint-disable-next-line func-names
+  httpMethods[prop] = function(url, config = undefined) {
+    return this({ ...config, method, url });
+  };
+}
+for (const prop of ['post', 'put', 'patch']) {
+  const method = prop.toUpperCase();
+  // eslint-disable-next-line func-names
+  httpMethods[prop] = function(url, data = undefined, config = undefined) {
+    const full = { ...config, method, url };
```

Comment:
what happens if data is null or undefined?

Reply:
Not specifying `data`, specifying `undefined` for `data`, and specifying `null` for `data` all have the same effect, which is that no data will be passed to axios. That's pretty rare, but it happens for POST /v1/assignments/:role/:actorId and POST /v1/projects/:projectId/assignments/:role/:actorId.

I suppose `null` is valid JSON, so it might make sense to distinguish between `null` and `undefined` here: I could just check for `undefined` below (`!== undefined` instead of `!= null`). I also think we could leave it as-is and change it if/when we need to POST a `null` value. Before they were removed, the convenience methods of the `request` mixin didn't distinguish between `null` and `undefined` values of `data`.

Reply:
I was thinking to merge these two for loops into one, if `if (data != null) full.data = data;` is not necessary.

Also `= undefined` is redundant in the function signature. By default it's `undefined` if consumer doesn't pass any value

Reply:
> I was thinking to merge these two for loops into one, if `if (data != null) full.data = data;` is not necessary.

The functions in the two loops have a different number of arguments, so I don't think they can be merged (unless I'm misunderstanding).

> Also `= undefined` is redundant in the function signature. By default it's `undefined` if consumer doesn't pass any value

It has some effect (on the `length` property of the function), but it's definitely not needed. I've gotten it in the habit of specifying `= undefined` as a way to signal that a parameter is optional.

Reply:
What is the implications of this?

```
const httpMethods = {};
for (const prop of ['get', 'delete', 'post', 'put', 'patch']) {
  const method = prop.toUpperCase();
  // eslint-disable-next-line func-names
  httpMethods[prop] = function(url, config, data) {
    return this({ ...config, method, url, data });
  };
}
```

Reply:
Ah I see what you mean. For POST/PUT/PATCH, I think it's nice to order `data` before `config`. Those methods will almost always receive `data`, but it's rare for them to receive `config`. `url, data, config` is also the same order as the axios convenience methods.

Reply:
right.